scip_indexer/SCIPIndexer.cc

```diff
@@ -1275,7 +1275,7 @@ class SCIPSemanticExtension : public SemanticExtension {
         // NOTE: We are not respecting the path prefix option here. Should we do that?
         // FIXME(varun): filesystem::current_path() returns the path in 'native' format,
         // so this won't work on Windows.
-        metadata.set_project_root("file:/" + filesystem::current_path().string());
+        metadata.set_project_root("file://" + filesystem::current_path().string());
```

Comment:
@varungandhi-src is the note above still actual?

Reply:
Yes, it's still accurate. Sorbet has an extra flag `remove-path-prefix` which is not accounted for here.

Reply:
So `scip-ruby` is broken on Windows? Not a big deal, just curious how C++ folks get around with that. Why ProjectRoot is URI at all and not a path?

Reply:
> So scip-ruby is broken on Windows?

It's not supported on Windows. We haven't tried to build it on Windows, and there is no release binary for Windows.

> Not a big deal, just curious how C++ folks get around with that.

Could you be a bit more specific here? It isn't clear to me what you're asking...

> Why ProjectRoot is URI at all and not a path?

Windows paths can be in many different formats. Using a URI avoids downstream tooling having to deal with that.

Reply:
> Could you be a bit more specific here? It isn't clear to me what you're asking...

How C++ folks are normalizing paths to be cross-platform.

Reply:
> How C++ folks are normalizing paths to be cross-platform.

Generally it isn't an operation that makes sense, because the set of allowed characters in path components is different. E.g. you can have a backslash inside a component on Linux but not on Windows IIUC. There's also the question of case-sensitivity which varies across filesystems (not OSes).

If one really wants, one can use string replacement or similar based on what the exact requirements are...

Reply:
That applies to most other fields of SCIP but not this one. For this particular field, it's not meant for the Sourcegraph backend, it's meant for any other tools in case the index is moved around/generated somewhere. E.g. the `scip snapshot` subcommand from the scip CLI can generate sources annotated with occurrence information -- it needs an absolute path to locate the source code.

Reply:
Leaking absolute project paths may be a security issue. Maybe I am just paranoid.